pandas/tests/io/formats/test_style.py

```diff
@@ -350,10 +350,10 @@ def test_bar_align_left(self):
             (0, 0): ['width: 10em', ' height: 80%'],
             (1, 0): ['width: 10em', ' height: 80%',
                      'background: linear-gradient('
-                     '90deg,#d65f5f 50.0%, transparent 0%)'],
+                     '90deg,#d65f5f 50.0%, transparent 50.0%)'],
```

Comment:
Why are these tests changing?

Reply:
The original implementation generated gradient stops that were invalid: the positions (percentages) should not decrease. In the example above the first stop (`#d65f5f`) is at 50% and de second (`transparent`) was originally at 0%, but it's more correct to have the second one explicitly at 50% as well. (Browsers are expected to guess the correct intention of these incorrect cases, but you never know of course.)

In other test cases, the original css was also bit redudant, e.g.:
```
transparent 0%, transparent 10.0%, #5fba7d 10.0%
```
can be simplified to 
```
transparent 10.0%, #5fba7d 10.0%
````

or even worse, sometimes things like these were part of the gradient: `transparent 0%, transparent 0.0%, #5fba7d 0.0%,` which is completely redundant.

The new implementation improves on all these points, so original tests were adapted.

Reply:
And there was also a test case (`test_bar_align_zero_pos_and_neg`) with bar size bug in the original implementation when `width < 100`, which is also fixed in new implementation